src/Value.hs

```diff
@@ -24,16 +24,22 @@ data MacroAction
   | MacroActionWaitSignal Signal
   | MacroActionIdentEq HowEq Value Value
   | MacroActionLog Syntax
+  | MacroActionIntroducer
   | MacroActionWhichProblem
-  deriving (Eq, Show)
+
+instance Show MacroAction where
+  show _ = "MacroAction..."
```

Comment:
Meh, if these Show instances aren't going to give any information, should we even bother giving them a Show instance at all?

Reply:
Because other things that include a MacroAction do have useful Show instances that are derived.